dataland-external-storage/src/main/kotlin/org/dataland/datalandexternalstorage/services/EurodatDataStore.kt

```diff
@@ -0,0 +1,272 @@
+import org.dataland.datalandeurodatclient.openApiClient.model.SafeDepositDatabaseResponse
+import org.dataland.datalandexternalstorage.entities.DataItem
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.dataland.datalandmessagequeueutils.constants.ActionType
+import org.dataland.datalandmessagequeueutils.constants.ExchangeName
+import org.dataland.datalandmessagequeueutils.constants.MessageHeaderKey
+import org.dataland.datalandmessagequeueutils.constants.MessageType
+import org.dataland.datalandmessagequeueutils.constants.RoutingKeyNames
+import org.dataland.datalandmessagequeueutils.exceptions.MessageQueueRejectException
+import org.dataland.datalandmessagequeueutils.utils.MessageQueueUtils
+import org.json.JSONObject
+import org.slf4j.LoggerFactory
+import org.springframework.amqp.rabbit.annotation.Argument
+import org.springframework.amqp.rabbit.annotation.Exchange
+import org.springframework.amqp.rabbit.annotation.Queue
+import org.springframework.amqp.rabbit.annotation.QueueBinding
+import org.springframework.amqp.rabbit.annotation.RabbitListener
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.messaging.handler.annotation.Header
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.sql.SQLException
+
+/**
+ * Simple implementation of a data storing service using the EuroDaT data trustee
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
+ * @param temporarilyCachedDataClient the service for retrieving data from the temporary storage
+ * @param temporarilyCachedDocumentClient the service for retrieven documents from the temporary storage
+ * @param objectMapper object mapper used for converting data classes to strings and vice versa
+ * @param databaseCredentialResourceClient the service to retrieve eurodat storage credentials
+ * @param safeDepositDatabaseResourceClient the service to create the safe deposit box used to store private data
+ * on eurodat
+ * @param messageUtils contains utils connected to the messages on the message queue
+ */
+@Suppress("LongParameterList")
+@Component
+class EurodatDataStore(
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired var temporarilyCachedDataClient: TemporarilyCachedDataControllerApi,
+    @Autowired var temporarilyCachedDocumentClient: StreamingTemporarilyCachedPrivateDocumentControllerApi,
+    @Autowired var databaseCredentialResourceClient: DatabaseCredentialResourceApi,
+    @Autowired var safeDepositDatabaseResourceClient: SafeDepositDatabaseResourceApi,
+    @Autowired var objectMapper: ObjectMapper,
```

Comment:
Von den Autowired Komponenten wird eine ganze Reihe gar nicht benutzt, oder übersehe ich hier etwas?

Reply:
Soweit ich gesehen habe werden die alle verwendet. Jedenfalls erhalte ich mehrere Treffer wenn ich nach ihnen in der Datei suche.